tools/c7n_azure/c7n_azure/resources/public_ip.py

```diff
@@ -33,3 +34,12 @@ class resource_type(object):
             'properties.publicIPAllocationMethod',
             'properties.ipAddress'
         )
+
+    def get_resources(self, resource_ids):
+        result = []
+        for resource_id in resource_ids:
+            resource_group = resource_id.split('/')[4]
+            name = resource_id.split('/')[8]
```

Comment:
We should probably find some place for utility methods, put one for getting both resource group and name from this string, and also update anywhere using them to use that.  We can just file a small issue/work item to do that.  (This we can just unit test this hacky thing we do everywhere in one place and it will be more readable everywhere)

Reply:
filed #2304